web/src/components/TaskHeader.test.tsx

```diff
@@ -192,6 +212,237 @@ describe("<TaskHeader />", () => {
+          legalStructureId:
+            randomInt() % 2 ? randomTradeNameLegalStructure() : randomPublicFilingLegalStructure(),
+        }),
+        taskProgress: {
+          [taxTaskId]: "COMPLETED",
+        },
+      });
+
+      renderTaskHeader(taxTask, userData);
+      fireEvent.click(screen.getByTestId("taskProgress"));
+      fireEvent.click(screen.getByTestId("IN_PROGRESS"));
+
+      expect(screen.getByText(Config.taxRegistrationModal.areYouSureTaxBody)).toBeInTheDocument();
+    });
+
+    it("updates the task progress if the user continues in the warning modal", async () => {
+      const taxTask = generateTask({ id: taxTaskId });
+      const userData = generateUserData({
+        profileData: generateProfileData({
+          businessName: "test-business",
+          legalStructureId:
+            randomInt() % 2 ? randomTradeNameLegalStructure() : randomPublicFilingLegalStructure(),
+        }),
+        taskProgress: {
+          [taxTaskId]: "COMPLETED",
+        },
+      });
+
+      renderTaskHeader(taxTask, userData);
+      fireEvent.click(screen.getByTestId("taskProgress"));
+      fireEvent.click(screen.getByTestId("IN_PROGRESS"));
+
+      expect(screen.getByText(Config.taxRegistrationModal.areYouSureTaxBody)).toBeInTheDocument();
+
+      fireEvent.click(screen.getByText(Config.taxRegistrationModal.areYouSureTaxContinueButton));
+      const dropdown = screen.getByTestId("taskProgress");
+      expect(within(dropdown).getByText(Config.taskProgress.IN_PROGRESS)).toBeInTheDocument();
+    });
+
+    it("doesn't update the task progress if the user cancels in the warning modal", async () => {
+      const taxTask = generateTask({ id: taxTaskId });
+      const userData = generateUserData({
+        profileData: generateProfileData({
+          businessName: "test-business",
+          legalStructureId:
+            randomInt() % 2 ? randomTradeNameLegalStructure() : randomPublicFilingLegalStructure(),
+        }),
+        taskProgress: {
+          [taxTaskId]: "COMPLETED",
+        },
+      });
+
+      renderTaskHeader(taxTask, userData);
+      fireEvent.click(screen.getByTestId("taskProgress"));
+      fireEvent.click(screen.getByTestId("IN_PROGRESS"));
+
+      expect(screen.getByText(Config.taxRegistrationModal.areYouSureTaxBody)).toBeInTheDocument();
+      fireEvent.click(screen.getByText(Config.taxRegistrationModal.areYouSureTaxCancelButton));
+      await waitFor(() => {
+        const dropdown = screen.getByTestId("taskProgress");
+        expect(within(dropdown).getByText(Config.taskProgress.COMPLETED)).toBeInTheDocument();
+      });
+      expect(userDataWasNotUpdated).toBeTruthy();
+    });
+
+    describe("when trade name legal structure", () => {
+      it("shows all fields except business name and tax field", () => {
+        const taxTask = generateTask({ id: taxTaskId });
+        const userData = generateUserData({
+          profileData: generateProfileData({
+            businessName: "",
```

Comment:
i'd delete this, because it doesn't impact the outcome of the tests right? we don't show businessName field regardless of this value